crates/turbo-trace/src/tracer.rs

```diff
@@ -61,11 +71,159 @@ impl Tracer {
         Self {
             files,
             ts_config,
-            source_map: Rc::new(SourceMap::default()),
+            cwd,
+            errors: Vec::new(),
+            source_map: Arc::new(SourceMap::default()),
         }
     }
 
-    pub fn trace(mut self, max_depth: Option<usize>) -> TraceResult {
+    pub async fn get_imports_from_file(
+        source_map: &SourceMap,
+        errors: &mut Vec<TraceError>,
+        resolver: &Resolver,
+        file_path: &AbsoluteSystemPath,
+    ) -> Option<(Vec<AbsoluteSystemPathBuf>, SeenFile)> {
+        // Read the file content
+        let Ok(file_content) = tokio::fs::read_to_string(&file_path).await else {
```

Comment:
Not worth blocking, but I wonder if throwing rayon or a thread pool would be more effective for the reverse trace. The only concurrency we're getting is the file reads and I'm not sure how much of the weight of this function is FS vs parsing and extracting info.

Reply:
yeah I'm not exactly sure. I think we should do some investigation after it gets some use